src/main/java/seedu/address/logic/commands/room/AllocateRoomCommand.java

```diff
@@ -26,90 +23,85 @@
             + "[" + PREFIX_PATIENT_NAME + "PATIENT NAME]\n"
             + "Example: " + COMMAND_WORD + " 23 "
-            + PREFIX_ROOM_NUMBER + "123 "
             + PREFIX_PATIENT_NAME + "Mary Doe";
 
-    public static final String MESSAGE_EDIT_ROOM_SUCCESS = "Edited Room: %1$s";
-    public static final String MESSAGE_ROOM_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_ALLOCATE_ROOM_SUCCESS = "Allocated Room: %1$s";
     public static final String MESSAGE_DUPLICATE_ROOM = "This room already exists in the application.";
 
-    private final Integer roomNumberToEdit;
-    private final EditRoomDescriptor editRoomDescriptor;
+    private final Integer roomNumberToAllocate;
+    private final AllocateRoomDescriptor allocateRoomDescriptor;
 
     /**
-     * Constructs an EditRoomCommand to edit the room with the room number {@code Integer}.
+     * Constructs an AllocateRoomCommand to edit the room with the room number {@code Integer}.
      *
-     * @param roomNumberToEdit room number to edit
-     * @param editRoomDescriptor details to edit the room with
+     * @param roomNumberToAllocate room number to edit
+     * @param allocateRoomDescriptor details to edit the room with
      */
-    public EditRoomCommand(Integer roomNumberToEdit, EditRoomDescriptor editRoomDescriptor) {
-        requireAllNonNull(roomNumberToEdit, editRoomDescriptor);
-        this.roomNumberToEdit = roomNumberToEdit;
-        this.editRoomDescriptor = new EditRoomDescriptor(editRoomDescriptor);
+    public AllocateRoomCommand(Integer roomNumberToAllocate, AllocateRoomDescriptor allocateRoomDescriptor) {
+        requireAllNonNull(roomNumberToAllocate, allocateRoomDescriptor);
+        this.roomNumberToAllocate = roomNumberToAllocate;
+        this.allocateRoomDescriptor = new AllocateRoomDescriptor(allocateRoomDescriptor);
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        ObservableList<Room> roomList = model.getRoomList();
-        Index index = model.checkIfRoomPresent(roomNumberToEdit);
+        Optional<Room> roomOptional = model.getRoomWithRoomNumber(roomNumberToAllocate);
 
-        if (index.getZeroBased() == 0) {
+        if (roomOptional.isEmpty()) {
             throw new CommandException(MESSAGE_INVALID_ROOM_NUMBER);
         }
 
-        Room roomToEdit = roomList.get(index.getZeroBased() - 1);
-        Room editedRoom = createEditedRoom(model, roomToEdit, editRoomDescriptor);
+        Room roomToAllocate = roomOptional.get();
+        Room roomWithAllocatedPatient = allocatePatientToRoom(model, roomToAllocate, allocateRoomDescriptor);
 
-        if (!roomToEdit.isSameRoom(editedRoom) && model.hasRoom(editedRoom)) {
+        if (!roomToAllocate.isSameRoom(roomWithAllocatedPatient) && model.hasRoom(roomWithAllocatedPatient)) {
             throw new CommandException(MESSAGE_DUPLICATE_ROOM);
```

Comment:
Should we change the Message_Duplicate_Room to make it clearer? I believe this message is sent when the patient is already allocated to the room and the user calls the command to allocate the same patient to the room again. Maybe the message should be "The patient is already allocated to this room." instead of "This room already exists..." to make it clearer?